File: pizzayolo/src/main/java/pizzayolo/entity/Adresse.java
package pizzayolo.entity;

import javax.persistence.Column;
import javax.persistence.Embeddable;

import com.fasterxml.jackson.annotation.JsonView;

@Embeddable
public class Adresse {
	
	@Column(name="numero")
	@JsonView(JsonViews.UtilisateurWithAdresse.class)
	private String numVoie;
	@Column(name="rue")
	@JsonView(JsonViews.UtilisateurWithAdresse.class)
	private String nomVoie;
	@Column(name="complement")
	@JsonView(JsonViews.UtilisateurWithAdresse.class)
	private String complement;
	@Column(name="code_postale")
	@JsonView(JsonViews.UtilisateurWithAdresse.class)
	private String cp;
	@Column(name="ville")
	@JsonView(JsonViews.UtilisateurWithAdresse.class)
	private String ville;

	public Adresse() {
	}

	public Adresse(String numVoie, String nomVoie, String complement, String cp, String ville) {
		this.numVoie = numVoie;
		this.nomVoie = nomVoie;
		this.complement = complement;
		this.cp = cp;
		this.ville = ville;
	}

	public String getNumVoie() {
		return numVoie;
	}

	public void setNumVoie(String numVoie) {
		this.numVoie = numVoie;
	}

	public String getNomVoie() {
		return nomVoie;
	}

	public void setNomVoie(String nomVoie) {
		this.nomVoie = nomVoie;
	}

	public String getComplement() {
		return complement;
	}

	public void setComplement(String complement) {
		this.complement = complement;
	}

	public String getCp() {
		return cp;
	}

	public void setCp(String cp) {
		this.cp = cp;
	}

	public String getVille() {
		return ville;
	}

	public void setVille(String ville) {
		this.ville = ville;
	}

}

File: pizzayoloangular/src/app/model/adresse.ts
export class Adresse {
  public constructor(
    private _numVoie?: string,
    private _nomVoie?: string,
    private _cp?: string,
    private _ville?: string
  ) {}

  get numVoie(): string | undefined {
    return this._numVoie;
  }

  set numVoie(value: string | undefined) {
    this._numVoie = value;
  }

  get nomVoie(): string | undefined {
    return this._nomVoie;
  }

  set nomVoie(value: string | undefined) {
    this._nomVoie = value;
  }

  get cp(): string | undefined {
    return this._cp;
  }

  set cp(value: string | undefined) {
    this._cp = value;
  }

  get ville(): string | undefined {
    return this._ville;
  }

  set ville(value: string | undefined) {
    this._ville = value;
  }
}
